### PR TITLE
Display credential offer

### PR DESCRIPTION
**Pull request by amanji**
# Summary of Changes

* Add full notification screen modals and navigation changes for credential offer workflow
  - Modals include a pending, credential received and credential arrived
  - Modal component is generic and reusable
* Refactors some navigation params into exported module
* Adds unit tests for credential offer workflow

# Related Issues

N/A

# Pull Request Checklist

This is just a reminder about the most common mistakes. Please make sure that you tick all _appropriate_ boxes. But please read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!

If an item doesn't apply to your pull request, **check it anyway** to make it apparent that there's nothing to do.

- [ ] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this).
- [ ] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components.
- [ ] Run prettier: `npm run style-format`
- [ ] Updated **documentation** for changed code and new or modified features.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

_PR template adapted from the Python attrs project._

### CODE REVIEW

File: App/components/modals/NotificationModal.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    flexDirection: 'row',
+    justifyContent: 'flex-end',
+  },
+  iconButton: {
+    padding: 20,
+    paddingVertical: 28,
+  },
+})
+
+interface Props {
```
Comment (jleach) [inline App/components/modals/NotificationModal.tsx]: Would it matter if these were named `NotificationModalProps` for typescripts help (autocomplete). Maybe it makes no differences if all props are typed as `Props`?

**Review by jleach — DISMISSED** [review]

File: App/components/modals/NotificationModal.tsx (hunk 1 above)
Comment (amanji) [reply App/components/modals/NotificationModal.tsx]: Yeah I think we should move to being more specific in terms of naming. I can update

File: App/components/modals/NotificationModal.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  title,
+  doneTitle,
+  onDone,
+  onHome,
+  visible,
+  children,
+}) => {
+  const { t } = useTranslation()
+  const navigation = useNavigation<StackNavigationProp<HomeStackParams>>()
+  const [modalVisible, setModalVisible] = useState<boolean>(true)
```
Comment (jleach) [inline App/components/modals/NotificationModal.tsx:56]: I wonder if you did `useState<boolean>(visible)` if you would need the `useEffect()` below?

**Review by jleach — APPROVED** [review]

File: App/components/modals/NotificationModal.tsx (hunk 2 above)
Comment (amanji) [reply App/components/modals/NotificationModal.tsx:56]: Hmm. I didn't think of that but I can give it a try and submit an update

**Pull request merged**